sdk/lib/_internal/js_runtime/lib/js_helper.dart

```diff
@@ -378,6 +378,14 @@ class Primitives {
     return result;
   }
 
+  static bool? parseBool(String source) {    
+    if (source == null) throw argumentErrorValue(source);
+    if (source.isEmpty) return null;
+    source = source.toLowerCase();
```

Comment:
I wouldn't allocate a new string to do this. It's ASCII, so case conversion is easy.

```dart
 static bool _compareIgnoreCase(String input, String lowerCaseTarget) {
   if (input.length != lowerCaseTarget.length) return false;
   for (var i = 0; i < input.length; i++) {
     if (input.codeUnitAt(i) | 0x20 != lowerCaseTarget.codeUnitAt(i)) {
       return false;
     }
   }
   return true;
 }

 static bool? parseBoolIgnoreCase(String source) =>
   _compareIgnoreCase(source, "true") || 
   (_compareIgnoreCase(source, "false") ? false : null);
```
The `compareIgnoreCase` bails out on the length if it's the other valid input, so it should be efficient.

Reply:
Ok, applied the fixes and tested.

![Screenshot_2](https://user-images.githubusercontent.com/15394508/213039305-a751da7d-ddd5-453e-9b49-574bfac1b14d.png)